### PR TITLE
Nexus worker support

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Nexus worker support to the Java SDK. This PR adds support for registering Nexus Services to Java SDK worker.

Note: This does **NOT** add support for _Nexus Operations -> Workflows_

### CODE REVIEW

File: temporal-sdk/src/main/java/io/temporal/worker/Worker.java
Comment (cretz) [inline temporal-sdk/src/main/java/io/temporal/worker/Worker.java:407]:
```suggestion
   * <p>An Nexus service object must be annotated with {@link
   * io.nexusrpc.handler.ServiceImpl}.
```

Technically unrelated to interfaces

File: temporal-sdk/src/main/java/io/temporal/worker/Worker.java
Comment (cretz) [inline temporal-sdk/src/main/java/io/temporal/worker/Worker.java:424]: To save me from digging too much, can you confirm no resources are used by a Nexus worker if no Nexus service impls are registered?

File: temporal-sdk/src/main/java/io/temporal/worker/WorkflowImplementationOptions.java
Comment (cretz) [inline temporal-sdk/src/main/java/io/temporal/worker/WorkflowImplementationOptions.java:139]: No need to update doc, but just for me to understand, map key is service name here not class name right (even though class name is default service name)?

**Review by cretz — APPROVED** [review]
This is great. I looked over quite a bit looking for any obvious flaws and couldn't find any. Tests look good (and saw #2215). Would like @bergundy review before merge though.

File: temporal-sdk/src/main/java/io/temporal/worker/Worker.java
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/worker/Worker.java:424]: Correct

**Review by bergundy — APPROVED** [review]
Looks great!

The only major comment I had was about payload to content conversion but we discussed that on Zoom and the approach you took is okay for now. Functionally it does what we want, we will likely want to prevent the serialization though and build better support for this in the Nexus Java SDK.

File: temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java:214]: I made them lower case to match my understanding of the spec https://github.com/nexus-rpc/api/blob/main/SPEC.md#operationinfo

File: temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java:147]: >Also, is it standard to add punctuations to error messages in our SDK?

Yes most do

File: temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java:247]: This is the executor to call the cancel the nexus operation context after its' timeout. By this point there shouldn't be any tasks to shutdown.

File: temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java
Comment (Quinn-With-Two-Ns) [reply temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java:140]: I do include the exceptions in this error, is there more we want to include?

**Comment by Quinn-With-Two-Ns** [comment]
Going to merge to unblock other PRs, will address any other comments in this PR in follow up PRs

File: temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java
Comment (bergundy) [reply temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java:140]: I see it's in the details, can we copy the message instead?

File: temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java
Comment (bergundy) [reply temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java:214]: I was suggesting that the enums in Java are made lower case so you don't have to do this transformation every time. Then the Nexus Java SDK would match the spec.

**Pull request merged**
